datastream-common/src/test/java/com/linkedin/datastream/metrics/TestMetrics.java

```diff
@@ -0,0 +1,92 @@
+/**
+ *  Copyright 2019 LinkedIn Corporation. All rights reserved.
+ *  Licensed under the BSD 2-Clause License. See the LICENSE file in the project root for license information.
+ *  See the NOTICE file in the project root for additional information regarding copyright ownership.
+ */
+package com.linkedin.datastream.metrics;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+/**
+ * Tests for {@link BrooklinMetrics}.
+ */
+public class TestMetrics {
+  @Test
+  public void testBrooklinMeterInfoComparison() {
```

Comment:
It would be great to assert for identical `hashCode()` values whenever we `assertEquals()`; this ensures we honor this part of the `hashCode()` contract:
```
If two objects are equal according to the equals(Object) method, then calling 
the hashCode method on each of the two objects must produce the same integer result.
```

Reply:
Good catch. There was a bug in hash code. Adding this helped in fixing it.